pandas/_libs/lib.pyx

```diff
@@ -678,6 +678,26 @@ def is_range_indexer(ndarray[int6432_t, ndim=1] left, Py_ssize_t n) -> bool:
     return True
 
 
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def is_range(ndarray[int6432_t, ndim=1] sequence, int64_t diff) -> bool:
```

Comment:
can this have a more verbose-but-descriptive name?  from this name alone id expect this to be a somehow-optimized `isinstance(sequence, range)`

Reply:
Sure thing. Renamed to `is_sequence_range`